### DISCUSSION TITLE
Infer fragment type without useFragment

### DISCUSSION THREAD

**Discussion by jeffreyyoung** [dotansimha/graphql-code-generator#8420] [General] [answered]
Is it possible to infer the type of a typed document node in pure typescript without calling `useFragment`??

for example I'd like something like this:
```tsx
import { Infer } from './gql/fragment-masking';

export const FilmFragment = graphql(/* GraphQL */ `
  fragment FilmItem on Film {
    id
    title
  }
`)

// I want to infer type without invoking useFragment
type InferredType = Infer<typeof FilmFragment>; 
// { id: string, title: string }!!

```

**Comment by charlypoly** [comment, marked as answer]
Hi @jeffreyyoung,

You can get the fragment type by using the `FragmentType<typeof FragmentDocument>` type helper, as shown in our docs: https://www.the-guild.dev/graphql/codegen/docs/guides/react-vue#writing-graphql-fragments
